website/files/models/osfstorage.py

```diff
@@ -107,15 +107,27 @@ def path(self):
     def is_checked_out(self):
         return self.checkout is not None
 
-    def delete(self, user=None, parent=None):
-        if self.node.preprint_file == self:
-            self.node._is_preprint_orphan = True
-            self.node.save()
+    @property
+    def _delete_allowed(self):
+        if self.node.preprint_file == self and not self.node._has_abandoned_preprint:
+            raise exceptions.FileNodeIsPrimaryFile()
         if self.is_checked_out:
             raise exceptions.FileNodeCheckedOutError()
-        return super(OsfStorageFileNode, self).delete(user=user, parent=parent)
+        return True
+
+    @property
+    def delete_allowed(self):
+        try:
+            return self._delete_allowed
+        except:
+            return False
+
+    def delete(self, user=None, parent=None):
+        return super(OsfStorageFileNode, self).delete(user=user, parent=parent) if self._delete_allowed else None
 
     def move_under(self, destination_parent, name=None):
+        if self.node.preprint_file == self and not self.node._has_abandoned_preprint:
+            raise exceptions.FileNodeIsPrimaryFile()
```

Comment:
From ticket:
>If the user attempts to move the file, the action will be a copy action

Does this ensure that behavior? It looks like it might prevent it